app/controllers/api/discussions_controller.rb

```diff
@@ -2,13 +2,8 @@ class API::DiscussionsController < API::RestfulController
   load_and_authorize_resource only: [:show, :mark_as_read, :set_volume]
   load_resource only: [:create, :update]
 
-  def discussions_for_dashboard
-    @discussions = page_collection discussions_for_preview
-    respond_with_discussions
-  end
-
-  def discussions_for_inbox
-    @discussions = discussions_for_preview('show_unread')
+  def dashboard
+    instantiate_collection { |collection| discussions_for_preview(collection) }
```

Comment:
This might be a little confusing at first glance, but we're passing a filtering block to instantiate collection here.

Then, when we instantiate the collection it looks a little like

- Get visible records
- filter those records if we're filtering things
- timeframe those records if we've got a timeframe
- page those records if we're paging things
- return the filtered, timeframed, and/or paged collection

I think it works rather nicely.